overlord/state/state.go

```diff
@@ -97,18 +97,21 @@ type State struct {
 	modified bool
 
 	cache map[interface{}]interface{}
+
+	pendingChangeByAttr map[string]func(*Change) bool
```

Comment:
Nitpick: maybe the key should be a typed string?

Reply:
we haven't introduced an attritube-key type for state entries so far, just string, it would be odd to start here